### ISSUE TITLE
rpmbuild hangs on futex in Debian 9

### ISSUE THREAD

**Issue by ddamw** [Microsoft/WSL#3233]
I am trying to build an RPM package using rpmbuild on Debian 9 (on Windows), but it hangs and eventually segfaults. 

Using strace, I found out it is trying to set up a futex and that hangs.

I used a tutorial for the simplest RPM possible, as described here: https://sites.google.com/site/syscookbook/rhel/rhel-rpm-build

I did a little research on futexes and learned that those were not easy to support. I am guessing rpmbuild uses futexes in a way not fully supported yet.

**Windows build**
Microsoft Windows [Version 10.0.17134.48]

**What I was doing**
`rpmbuild -ba rpmbuild/SPECS/HelloWorld-1.0.spec`

**What is wrong**
The process hangs for a long time and eventually segfaults

**strace**
Relevant part:
```
open("/home/redacted/.rpmdb/__db.003", O_RDWR|O_CREAT, 0644) = 6
fcntl(6, F_GETFD)                       = 0
fcntl(6, F_SETFD, FD_CLOEXEC)           = 0
mmap(NULL, 1318912, PROT_READ|PROT_WRITE, MAP_SHARED, 6, 0) = 0x7fe40def0000
futex(0x7fe412860370, FUTEX_WAIT, 2, NULL
```
Please notice the unclosed function call at the end.

full strace: https://gist.github.com/ddamw/7a9ecb5b1b68f3cae6aeecaa616a8b6a

**scripts**
Issue the following commands in your home folder:
```
mkdir -p ~/rpmbuild/{BUILD,BUILDROOT,RPMS,SOURCES,SPECS,SRPMS}
echo '%_topdir %(echo $HOME)/rpmbuild' > ~/.rpmmacros
mkdir -p ~/rpmbuild/SOURCES/HelloWorld-1.0
echo "echo Hello World" > ~/rpmbuild/SOURCES/HelloWorld-1.0/HelloWorld.sh
cd rpmbuild/SOURCES
chmod 755 HelloWorld-1.0/HelloWorld.sh
touch HelloWorld-1.0/configure
chmod 755 HelloWorld-1.0/configure
tar czvf HelloWorld-1.0.tar.gz HelloWorld-1.0
cd
vi ~/rpmbuild/SPECS/HelloWorld-1.0.spec
```

Insert the following in the file opened above:
```
Name: HelloWorld
Version: 1.0
Release: 1%{?dist}
Summary: Hello World Script
Group: Miscellaneous
License: Licence text
Source0: HelloWorld-1.0.tar.gz
BuildArch: noarch
BuildRoot: %{_tmppath}/%{name}-%{version}-%{release}-root-%(%{__id_u} -n)

%description
This is a text describing what the Package is meant for

%prep

%setup -q

%build

%install

rm -rf $RPM_BUILD_ROOT
install -d -m 0755 $RPM_BUILD_ROOT/opt/HelloWorld
install -m 0755 HelloWorld.sh $RPM_BUILD_ROOT/opt/HelloWorld/HelloWorld.sh

%clean

rm -rf $RPM_BUILD_ROOT

%files

%defattr(-,root,root,-)

# %doc

/opt/HelloWorld/HelloWorld.sh

%changelog
```

Now start the build using:
`rpmbuild -ba rpmbuild/SPECS/HelloWorld-1.0.spec`

**Comment by therealkenc** [comment]
This one is a restatement of #2852 with a supported platform and repro steps (I haven't run it yet). That "relevant part" fail isn't going to be the real problem. The `__db.003` (a Berkeley DB) is almost certainly corrupt at that point, and that `futux()` address is likely pointing into space. That last step in the repro can be tightened if it is converted from a cut and paste to a `wget`/`curl` of a gist. Might improve the chances of someone running it. Also be worth trying the same sequence on Ubuntu 16.04 and see if it fails there too. But good submission on the merits 👍.

**Comment by ddamw** [comment]
You're right, the problem is not the futex, but a corrupted RPM database. I removed the  .rpmdb directory completely and reran the command and now it segfaults almost immediately while working on one of those files.

I guess this bug report can be closed then, as it is another problem.

For reference: 
I created a script from the steps above which anyone can download and run: https://gist.githubusercontent.com/ddamw/7a9ecb5b1b68f3cae6aeecaa616a8b6a/raw/aabc3509928f23ca1dba0f4cfbe64a44e4e5275c/rpmbuildtest.sh
I also ran this script on Ubuntu with a similar result: corrupted database and an almost eternal wait on subsequent runs.
